alfalfa_worker/jobs/openstudio/step_run.py

```diff
@@ -31,340 +43,304 @@ def __init__(self, run_id, realtime, timescale, external_clock, start_datetime,
+        for alfalfa_json in self.run.dir.glob('**/run/alfalfa.json'):
+            self.variables.generate_points(alfalfa_json)
 
-        :return:
-        """
-        self.osm_idf_files_prep()
-        (self.ep.status, self.ep.msg) = self.ep.start()
-        if self.ep.status != 0:
-            raise JobExceptionExternalProcess('Could not start EnergyPlus: {}'.format(self.ep.msg))
-
-        start_time = time()
+        self.ep_api: EnergyPlusAPI = None
+        self.ep_state = None
 
-        while time() < start_time + self.model_start_timeout and not self.ep.is_running:
-            self.check_error_log()
+        self.additional_points: list[AlfalfaPoint] = []
 
-            try:
-                [self.ep.status, self.ep.msg] = self.ep.accept_socket()
-            except socket.timeout:
-                pass
-
-        if not self.ep.is_running:
-            raise JobExceptionExternalProcess('Timedout waiting for EnergyPlus')
-
-        self.set_run_time(self.start_datetime)
-
-    def advance_to_start_time(self):
-        """ We get near the requested start time by manipulating the idf file,
-        however the idf start time is limited to the resolution of 1 day.
-        The purpose of this function is to move the simulation to the requested
-        start minute.
-        This is accomplished by advancing the simulation as quickly as possible. Data is not
-        published to database during this process
-        """
-        self.update_outputs_from_ep()
-
-        current_ep_time = self.get_sim_time()
-        self.logger.info(
-            'current_ep_time: {}, start_datetime: {}'.format(current_ep_time, self.start_datetime))
-        while True:
-            current_ep_time = self.get_sim_time()
-            if current_ep_time < self.start_datetime:
-                self.logger.info(
-                    'current_ep_time: {}, start_datetime: {}'.format(current_ep_time, self.start_datetime))
-                self.step()
-            else:
-                self.logger.info(f"current_ep_time: {current_ep_time} reached desired start_datetime: {self.start_datetime}")
-                break
-        self.master_enable_bypass = False
-        self.update_db()
-
-    def update_outputs_from_ep(self):
-        """Reads outputs from E+ step"""
-        self.check_error_log()
-        packet = self.ep.read()
-
-        _flag, _, outputs = mlep.mlep_decode_packet(packet)
-        if _flag == -1:
-            self.check_error_log()
-            raise JobExceptionExternalProcess("EnergyPlus experienced unknown error")
-        elif _flag == -10:
-            self.check_error_log()
-            raise JobExceptionExternalProcess("EnergyPlus experienced initialization error")
-        elif _flag == -20:
-            self.check_error_log()
-            raise JobExceptionExternalProcess("EnergyPlus experienced time integration error")
-
-        self.ep.outputs = outputs
-
-    def step(self):
+    def start_simulation_process(self):
```

Comment:
and critically, this functions blocks while EnergyPlus is running so it is started in another Process. You might make a note about that.